src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -325,10 +373,90 @@ internal static bool ValidateParentTagName(string parentTag, ErrorSink errorSink
                     errorSink: errorSink);
         }
 
-        private static bool ValidateName(
-            string name,
-            bool targetingAttributes,
-            ErrorSink errorSink)
+        private static bool ValidateRequiredAttribute(string attribute, ErrorSink errorSink)
+        {
+            if (attribute.Length >= 1 && attribute[0] == '[')
+            {
+                // Value starts with '[', treat it as a CSS selector.
+
+                if (attribute[attribute.Length - 1] != ']')
+                {
+                    errorSink.OnError(SourceLocation.Zero, "TODO: Create error for invalid CSS Selector syntax", length: 0);
+                    return false;
+                }
+
+                var equalsIndex = attribute.IndexOf('=');
+                if (equalsIndex != -1)
+                {
+                    var emptyValueSelector = false;
+
+                    // Don't need to bounds check here because the ']' error validation ensures that there's at least 1
+                    // extra character after the '='.
+                    var afterEquals = attribute[equalsIndex + 1];
+                    if ((afterEquals == '\'' || afterEquals == '"'))
+                    {
+                        // Check if there's a separate matching quote at the end of the attribute selector.
+                        if (attribute.Length - 2 != equalsIndex + 1 &&
+                            attribute[attribute.Length - 2] != afterEquals)
+                        {
+                            errorSink.OnError(SourceLocation.Zero, "TODO: Create error for unbalanced quotes for CSS Selector.", length: 0);
+                            return false;
+                        }
+                        else if (equalsIndex + 2 == attribute.Length - 2)
+                        {
+                            emptyValueSelector = true;
+                        }
+                    }
+                    else if (equalsIndex + 1 == attribute.Length -1)
+                    {
+                        emptyValueSelector = true;
+                    }
+
+                    if (emptyValueSelector)
+                    {
+                        errorSink.OnError(SourceLocation.Zero, "TODO: Create error for empty attribute selector value.", length: 0);
```

Comment:
`[name=""]` should absolutely be valid. How else would tag helpers target elements with `name=""`? Further the CSS spec does not disallow empty values for any attribute selector. (It's merely that a few never match.)